internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -459,7 +459,7 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 					Schema: map[string]*pluginsdk.Schema{
 						"dns_zone_id": {
 							Type:         pluginsdk.TypeString,
-							Required:     true,
+							Optional:     true,
 							ValidateFunc: dnsValidate.ValidateDnsZoneID,
```

Comment:
Instead of setting this to Optional and having to supply an empty block to enable this (which we don't do for any feature in this resource) I think the validation should allow an empty string as a valid value and the property description in the docs updated.